### DISCUSSION TITLE
对于含有大量katex的长文本如何优化

### DISCUSSION THREAD

**Discussion by D-Sketon** [theme-shoka-x/hexo-theme-shokaX#40] [Q&A] [answered]
比如50k个字的博客，烟花和头图动画卡的不行（）

**Comment by zkz098** [comment, marked as answer]
拆成多篇文章可能是个好方法()
有没有md文件或者选段，可以私发到我邮箱(`zkz20081204#126.com`)，我做个测试

**Reply by D-Sketon** [reply]
已发

**Reply by zkz098** [reply]
目前看来最好的方法就是拆成多篇文章
这个量的Katex公式优化较为困难，在6x CPU slowdown下整个页面和卡死了一样()
正在尝试通过懒加载等方式优化，但是进展缓慢

**Reply by zkz098** [reply]
找到了一个方法，测试环境下速度好了很多，很快就发测试版

**Reply by zkz098** [reply]
安装一下0.2.0-rc2试试，用`hexo-theme-shokax@dev`即可 @D-Sketon

**Reply by D-Sketon** [reply]
> 安装一下0.2.0-rc2试试，用`hexo-theme-shokax@dev`即可 @D-Sketon 

等明天考完试来测试一下()

**Reply by D-Sketon** [reply]
> 安装一下0.2.0-rc2试试，用`hexo-theme-shokax@dev`即可 @D-Sketon

确实流畅了不少，但上下滚动时上方的导航栏会抽搐

**Comment by zkz098** [comment]
导航栏抽搐问题目前还没法解决，因为这个优化方法会导致页面的`window.scrollY`反复抖动，右下角的返回顶部按钮也有类似问题。
给ScrollHandle加防抖会导致导航栏响应速度很慢

**Reply by D-Sketon** [reply]
content-visibility有点鸡肋哇

**Reply by zkz098** [reply]
但content-visibility是最有用的，本质还是shoka的scroll处理方面有问题，在测试markdown里面scrollHandle运行一次需要120ms左右。用content-visibility暴力缩短渲染范围是目前最简单的方法了

**Reply by D-Sketon** [reply]
还是contain-intrinsic-size没法准确获取吧

**Reply by zkz098** [reply]
contain-intrinsic-size本身就是估算，不可能准确啊（
所以抖动问题目前只能减少，很难根除。0.2.0-rc4将此项功能更改为可选功能
